### PR TITLE
[llm benchmark] Add qlora  benchmark

### PR DESCRIPTION
**Pull request by w5688414**
### Description

| Model     | Method   | Num GPUs | Batch Size  | Paddle Setup | Paddle Effective Tokens/s | Torch Setup | Torch Effective Tokens/s | Speedup |
|-----------|----------|----------|-------------|--------------|---------------------------|-------------|--------------------------|---------|
| Llama-65b | LoRA     | 4        | 8           | MP4 recompute|  474.36                   | gradient ckpt, bits 4, max_memory_MB 50000, qlora        | 327.75          |  **45%** |
| Llama-65b | LoRA     | 4        | 16          | MP4 recompute|  452.20                   | gradient ckpt, bits 4, max_memory_MB 50000, qlora        | 405.90          |  **11%** |

+ qlora
```
export CUDA_VISIBLE_DEVICES=5
python benchmark.py --model_name_or_path  llama-65b-hf-pt \
    --num_train_epochs 1 --per_device_train_batch_size 4 \
    --evaluation_strategy no --save_strategy no \
    --gradient_checkpointing \
    --fp16 \
    --fp16_opt_level O2 \
    --bits 4 \
    --lora \
    --tf32 False  \
    --fp16_full_eval \
    --logging_steps 50 --output_dir outputs
```

```
export CUDA_VISIBLE_DEVICES=0,1,2,3,4,5,6,7
python -m torch.distributed.run --nproc_per_node=8 benchmark.py \
    --model_name_or_path  llama-65b-hf-pt \
    --num_train_epochs 1 --per_device_train_batch_size 1 \
    --gradient_checkpointing \
    --evaluation_strategy no --save_strategy no \
    --fp16 \
    --qlora \
    --max_memory_MB 50000 \
    --bits 4 \
    --logging_steps 50 --output_dir outputs
```

### CODE REVIEW

File: examples/benchmark/peft/paddle/benchmark.py [hunk 1]
```diff
@@ -131,45 +137,48 @@ def preprocess_function(example, max_src_length=256, max_tgt_length=384):
...
-    
+
     if model_args.english:
         dataset = load_dataset("tatsu-lab/alpaca")
     else:
         dataset = load_dataset("Chinese-Vicuna/guanaco_belle_merge_v1.0")
 
     # select first 10k examples for benchmarking
-    dataset = dataset["train"].select(range(10000))
+    dataset = dataset["train"].select(range(1000))
```
Comment (sijunhe) [inline examples/benchmark/peft/paddle/benchmark.py]: 这个配一个参数吧

File: examples/benchmark/peft/torch/benchmark.py [hunk 2]
```diff
@@ -58,31 +57,79 @@ class ModelArguments:
...
 
     if "llama" in model_args.model_name_or_path:
-        tokenizer = LlamaTokenizer.from_pretrained(model_args.model_name_or_path,
-                                                    use_fast=False)
+        tokenizer = LlamaTokenizer.from_pretrained(model_args.model_name_or_path, use_fast=False)
         tokenizer.pad_token_id = 0
     elif "chatglm" in model_args.model_name_or_path:
         tokenizer = AutoTokenizer.from_pretrained(model_args.model_name_or_path, trust_remote_code=True)
     else:
         tokenizer = AutoTokenizer.from_pretrained(model_args.model_name_or_path)
```
Comment (sijunhe) [inline examples/benchmark/peft/torch/benchmark.py]: 2个branch统一都用trust_remote_code=True应该是可以的吧？

**Review by sijunhe — COMMENTED** [review]
pad to max length以及DataCollatorForSupervisedDataset 还是移出掉吧

File: examples/benchmark/peft/paddle/benchmark.py (hunk 1 above)
Comment (w5688414) [reply examples/benchmark/peft/paddle/benchmark.py]: 已修改

File: examples/benchmark/peft/torch/benchmark.py (hunk 2 above)
Comment (w5688414) [reply examples/benchmark/peft/torch/benchmark.py]: 已修改

**Review by sijunhe — APPROVED** [review]
lgtm

**Pull request merged**